### PR TITLE
Fix proximity checkers not being destroyed

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
`monitor.checkers` is a lazy list, this was runtiming.

lol bee code

## Why It's Good For The Game
vlc_2021-05-09T01-38-09

## Changelog
:cl:
fix: Proximity checkers, used in the peacekeeper borg module, will now properly destroy.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/proximity.dm [hunk 1]
```diff
@@ -118,7 +118,8 @@
 	return
 
 /obj/effect/abstract/proximity_checker/Destroy()
-	monitor.checkers -= src
+	if (monitor.checkers)
+		monitor.checkers -= src
```
Comment (LemonInTheDark) [inline code/game/objects/effects/proximity.dm:122]: if this is a lazylist you should use the macros, even if you hate them they at least prevent people from being dumb downline right?

Comment (Mothblocks) [reply code/game/objects/effects/proximity.dm:122]: sure theyre just ugly

Comment (Mothblocks) [reply code/game/objects/effects/proximity.dm:122]: watch this

```suggestion
	LAZYREMOVE(monitor.checkers, src)
```

Comment (LemonInTheDark) [reply code/game/objects/effects/proximity.dm]: based?

**Review by Timberpoes — APPROVED** [review]

**Review by LemonInTheDark — APPROVED** [review]

**Comment by ike709** [comment]
> lol bee code

We didn't merge it 😎

**Comment by Archanial** [comment]
This will runtime because when you QDEL_LIST and remove the last item from it LAZYREMOVE will null it, which makes it produce null.Cut() runtime.
https://github.com/BeeStation/BeeStation-Hornet/pull/4319

**Comment by LemonInTheDark** [comment]
Well it shouldn't be QDEL_LIST()'d then yeah

**Pull request merged**